Failure with debug output:

```
envoy> ERROR: /build/output/external/com_google_protobuf/src/google/protobuf/BUILD.bazel:548:11: Compiling src/google/protobuf/map_field.cc [for tool] failed: (Exit 1): gcc failed: error executing command (from target @com_google_protobuf//src/google/protobuf:protobuf)
envoy>   (cd /build/output/execroot/envoy && \
envoy>   exec env - \
envoy>     CC=gcc \
envoy>     CXX=g++ \
envoy>     PATH=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin:/nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin:/nix/store/00g69vw7c9lycy63h45ximy0wmzqx5y6-diffutils-3.10/bin:/nix/store/ks6xg51nxksqy772bj3y5682m0xqa689-file-5.46/bin:/nix/store/jqrz1vq5nz4lnv9pqzydj0ir58wbjfy1-findutils-4.10.0/bin:/nix/store/a3c47r5z1q2c4rz0kvq8hlilkhx2s718-gawk-5.3.1/bin:/nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin:/nix/store/5yja5dpk2qw1v5mbfbl2d7klcdfrh90w-patch-2.7.6/bin:/nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin:/nix/store/9cwwj1c9csmc85l2cqzs3h9hbf1vwl6c-gnutar-1.35/bin:/nix/store/nvvj6sk0k6px48436drlblf4gafgbvzr-gzip-1.13/bin:/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/bin:/nix/store/mpncgrbpnsrh62qplwzr6mcnq1by79rn-unzip-6.0/bin:/nix/store/4v2jp43ag2kynby75pnpvy34c3cbd18j-which-2.21/bin:/nix/store/yd6077p26k2i20ys27l2lifj6fxqszyx-zip-3.0/bin:/nix/store/38cffsqqx823crf1i4bcf6zz1qz1hgpd-cmake-3.30.5/bin:/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/bin:/nix/store/xp9q8zlbbrwp2mifh42hwsjpz9k4ag5l-gn-unstable-2024-05-13/bin:/nix/store/jfv85qbj4vb1dafcg6kncg4vrbq2bbxv-go-1.23.4/bin:/nix/store/f1sx0rkgn7zs5h6rvhavwpbw5896cn9y-openjdk-headless-11.0.25+9/bin:/nix/store/byvgvqlh007flwk940c99zfgzrjg0zsg-ninja-1.12.1/bin:/nix/store/srfxqk119fijwnprgsqvn68ys9kiw0bn-patchelf-0.15.0/bin:/nix/store/mz3infp9wpr1bzi6v550yl0c8i5bdf68-bazel-hacks-6.5.0/bin:/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin:/nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/bin:/nix/store/1c6bmxrrhm8bd26ai2rjqld2yyjrxhds-glibc-2.40-36-bin/bin:/nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin:/nix/store/srcmmqi8kxjfygd0hyy42c8hv6cws83b-binutils-wrapper-2.43.1/bin:/nix/store/j7p46r8v9gcpbxx89pbqlh61zhd33gzv-binutils-2.43.1/bin:/nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin:/nix/store/jqrz1vq5nz4lnv9pqzydj0ir58wbjfy1-findutils-4.10.0/bin:/nix/store/00g69vw7c9lycy63h45ximy0wmzqx5y6-diffutils-3.10/bin:/nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin:/nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin:/nix/store/a3c47r5z1q2c4rz0kvq8hlilkhx2s718-gawk-5.3.1/bin:/nix/store/9cwwj1c9csmc85l2cqzs3h9hbf1vwl6c-gnutar-1.35/bin:/nix/store/nvvj6sk0k6px48436drlblf4gafgbvzr-gzip-1.13/bin:/nix/store/mglixp03lsp0w986svwdvm7vcy17rdax-bzip2-1.0.8-bin/bin:/nix/store/fp6cjl1zcmm6mawsnrb5yak1wkz2ma8l-gnumake-4.4.1/bin:/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin:/nix/store/5yja5dpk2qw1v5mbfbl2d7klcdfrh90w-patch-2.7.6/bin:/nix/store/h18s640fnhhj2qdh5vivcfbxvz377srg-xz-5.6.3-bin/bin:/nix/store/c4rj90r2m89rxs64hmm857mipwjhig5d-file-5.46/bin:/nix/store/ks6xg51nxksqy772bj3y5682m0xqa689-file-5.46/bin:/nix/store/mpncgrbpnsrh62qplwzr6mcnq1by79rn-unzip-6.0/bin:/nix/store/4v2jp43ag2kynby75pnpvy34c3cbd18j-which-2.21/bin:/nix/store/yd6077p26k2i20ys27l2lifj6fxqszyx-zip-3.0/bin \
envoy>     PWD=/proc/self/cwd \
envoy>   /nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections '-std=c++14' -MD -MF bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_objs/protobuf/map_field.d '-frandom-seed=bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_objs/protobuf/map_field.o' '-DBAZEL_CURRENT_REPOSITORY="com_google_protobuf"' -iquote external/com_google_protobuf -iquote bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf -iquote external/com_google_absl -iquote bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_absl -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/internal_visibility -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/port -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena_align -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/stubs/_virtual_includes/lite -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena_allocation_policy -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena_cleanup -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/string_block -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/varint_shuffle -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/io -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/io_win32 -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/third_party/utf8_range/_virtual_includes/utf8_validity -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/third_party/utf8_range/_virtual_includes/utf8_range -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/gzip_stream -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/stubs/_virtual_includes/stubs -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/printer -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/zero_copy_sink -Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/tokenizer -isystem bazel-out/k8-opt-exec-2B5CBBC6/bin/bazel/foreign_cc/zlib/include -g0 '-frandom-seed=5aycnjj3bc' -isystem /nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/include -isystem /nix/store/f1sx0rkgn7zs5h6rvhavwpbw5896cn9y-openjdk-headless-11.0.25+9/include -isystem /nix/store/pr0rrwgi9fxxkk31628bw8wqfwj66fj2-linux-headers-6.12/include -isystem /nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/include -isystem /nix/store/f1sx0rkgn7zs5h6rvhavwpbw5896cn9y-openjdk-headless-11.0.25+9/include -isystem /nix/store/pr0rrwgi9fxxkk31628bw8wqfwj66fj2-linux-headers-6.12/include -g0 '-std=c++20' -fsized-deallocation -DHAVE_ZLIB -Woverloaded-virtual -Wno-sign-compare -Wno-nonnull -Werror -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/com_google_protobuf/src/google/protobuf/map_field.cc -o bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_objs/protobuf/map_field.o)
envoy> # Configuration: 75d4680c85066195ead4438447700f5a0b47c8c5016806f59e82c1eaaacb1e2a
envoy> # Execution platform: @local_config_platform//:host
envoy> In file included from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/string:54,
envoy>                  from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/locale_classes.h:40,
envoy>                  from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/ios_base.h:41,
envoy>                  from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/streambuf:43,
envoy>                  from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/streambuf_iterator.h:35,
envoy>                  from /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/iterator:66,
envoy>                  from external/com_google_absl/absl/synchronization/mutex.h:64,
envoy>                  from bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:15,
envoy>                  from external/com_google_protobuf/src/google/protobuf/map_field.cc:8:
envoy> In member function 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::pointer std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_data() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]',
envoy>     inlined from 'constexpr bool std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_is_local() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:269:13,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::capacity() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1182:20,
envoy>     inlined from 'constexpr void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.tcc:283:41,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1619:17,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator=(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:819:21,
envoy>     inlined from 'void google::protobuf::MapKey::CopyFrom(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:205:74,
envoy>     inlined from 'google::protobuf::MapKey::MapKey(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:65:51,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorageInternal(T*, google::protobuf::Arena*, std::false_type, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:617:5,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorage(T*, google::protobuf::Arena*, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:602:33,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::TryEmplaceInternal(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1605:34,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::ArenaAwareTryEmplace(std::false_type, Args&& ...) [with Args = {const google::protobuf::MapKey&, const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1656:30,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::try_emplace(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1430:32,
envoy>     inlined from 'void google::protobuf::Map<Key, T>::insert(InputIt, InputIt) [with InputIt = google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::const_iterator; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1452:18,
envoy>     inlined from 'google::protobuf::Map<Key, T>& google::protobuf::Map<Key, T>::operator=(const google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1500:13:
envoy> /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:228:28: error: '*(const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)((char*)<unknown> + 8).std::__cxx11::basic_string<char>::_M_dataplus.std::__cxx11::basic_string<char>::_Alloc_hider::_M_p' may be used uninitialized [-Werror=maybe-uninitialized]
envoy>   228 |       { return _M_dataplus._M_p; }
envoy>       |                            ^~~~
envoy> In member function 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::capacity() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]',
envoy>     inlined from 'constexpr void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.tcc:283:41,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1619:17,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator=(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:819:21,
envoy>     inlined from 'void google::protobuf::MapKey::CopyFrom(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:205:74,
envoy>     inlined from 'google::protobuf::MapKey::MapKey(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:65:51,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorageInternal(T*, google::protobuf::Arena*, std::false_type, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:617:5,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorage(T*, google::protobuf::Arena*, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:602:33,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::TryEmplaceInternal(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1605:34,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::ArenaAwareTryEmplace(std::false_type, Args&& ...) [with Args = {const google::protobuf::MapKey&, const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1656:30,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::try_emplace(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1430:32,
envoy>     inlined from 'void google::protobuf::Map<Key, T>::insert(InputIt, InputIt) [with InputIt = google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::const_iterator; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1452:18,
envoy>     inlined from 'google::protobuf::Map<Key, T>& google::protobuf::Map<Key, T>::operator=(const google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1500:13:
envoy> /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1182:30: error: '*(const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)((char*)<unknown> + 8).std::__cxx11::basic_string<char>::<anonymous>.std::__cxx11::basic_string<char>::<unnamed union>::_M_allocated_capacity' may be used uninitialized [-Werror=maybe-uninitialized]
envoy>  1182 |         return _M_is_local() ? size_type(_S_local_capacity)
envoy>       |                ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
envoy>  1183 |                              : _M_allocated_capacity;
envoy>       |                              ~~~~~~~~~~~~~~~~~~~~~~~
envoy> In member function 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::pointer std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_data() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]',
envoy>     inlined from 'constexpr bool std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_is_local() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:269:13,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::capacity() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1182:20,
envoy>     inlined from 'constexpr void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.tcc:283:41,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1619:17,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator=(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:819:21,
envoy>     inlined from 'void google::protobuf::MapKey::CopyFrom(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:205:74,
envoy>     inlined from 'google::protobuf::MapKey::MapKey(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:65:51,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorageInternal(T*, google::protobuf::Arena*, std::false_type, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:617:5,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorage(T*, google::protobuf::Arena*, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:602:33,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::TryEmplaceInternal(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1605:34,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::ArenaAwareTryEmplace(std::false_type, Args&& ...) [with Args = {const google::protobuf::MapKey&, const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1656:30,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::try_emplace(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1430:32,
envoy>     inlined from 'void google::protobuf::Map<Key, T>::insert(InputIt, InputIt) [with InputIt = google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::const_iterator; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1452:18,
envoy>     inlined from 'google::protobuf::Map<Key, T>::Map(google::protobuf::Arena*, const google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1192:11,
envoy>     inlined from 'google::protobuf::Map<Key, T>::Map(const google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1146:45,
envoy>     inlined from 'void google::protobuf::Map<Key, T>::swap(google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1512:11,
envoy>     inlined from 'static void google::protobuf::internal::TypeDefinedMapFieldBase<Key, T>::SwapImpl(google::protobuf::internal::MapFieldBase&, google::protobuf::internal::MapFieldBase&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field_inl.h:133:55:
envoy> /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:228:28: error: '*(const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)((char*)<unknown> + 8).std::__cxx11::basic_string<char>::_M_dataplus.std::__cxx11::basic_string<char>::_Alloc_hider::_M_p' may be used uninitialized [-Werror=maybe-uninitialized]
envoy>   228 |       { return _M_dataplus._M_p; }
envoy>       |                            ^~~~
envoy> In member function 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::capacity() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]',
envoy>     inlined from 'constexpr void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.tcc:283:41,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1619:17,
envoy>     inlined from 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator=(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]' at /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:819:21,
envoy>     inlined from 'void google::protobuf::MapKey::CopyFrom(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:205:74,
envoy>     inlined from 'google::protobuf::MapKey::MapKey(const google::protobuf::MapKey&)' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field.h:65:51,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorageInternal(T*, google::protobuf::Arena*, std::false_type, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:617:5,
envoy>     inlined from 'static void google::protobuf::Arena::CreateInArenaStorage(T*, google::protobuf::Arena*, Args&& ...) [with T = google::protobuf::MapKey; Args = {const google::protobuf::MapKey&}]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/arena.h:602:33,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::TryEmplaceInternal(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1605:34,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::ArenaAwareTryEmplace(std::false_type, Args&& ...) [with Args = {const google::protobuf::MapKey&, const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1656:30,
envoy>     inlined from 'std::pair<google::protobuf::Map<Key, T>::iterator, bool> google::protobuf::Map<Key, T>::try_emplace(K&&, Args&& ...) [with K = const google::protobuf::MapKey&; Args = {const google::protobuf::MapValueRef&}; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1430:32,
envoy>     inlined from 'void google::protobuf::Map<Key, T>::insert(InputIt, InputIt) [with InputIt = google::protobuf::Map<google::protobuf::MapKey, google::protobuf::MapValueRef>::const_iterator; Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1452:18,
envoy>     inlined from 'google::protobuf::Map<Key, T>::Map(google::protobuf::Arena*, const google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1192:11,
envoy>     inlined from 'google::protobuf::Map<Key, T>::Map(const google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1146:45,
envoy>     inlined from 'void google::protobuf::Map<Key, T>::swap(google::protobuf::Map<Key, T>&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/map.h:1512:11,
envoy>     inlined from 'static void google::protobuf::internal::TypeDefinedMapFieldBase<Key, T>::SwapImpl(google::protobuf::internal::MapFieldBase&, google::protobuf::internal::MapFieldBase&) [with Key = google::protobuf::MapKey; T = google::protobuf::MapValueRef]' at bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf/google/protobuf/map_field_inl.h:133:55:
envoy> /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/basic_string.h:1182:30: error: '*(const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)((char*)<unknown> + 8).std::__cxx11::basic_string<char>::<anonymous>.std::__cxx11::basic_string<char>::<unnamed union>::_M_allocated_capacity' may be used uninitialized [-Werror=maybe-uninitialized]
envoy>  1182 |         return _M_is_local() ? size_type(_S_local_capacity)
envoy>       |                ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
envoy>  1183 |                              : _M_allocated_capacity;
envoy>       |                              ~~~~~~~~~~~~~~~~~~~~~~~
envoy> cc1plus: all warnings being treated as errors
envoy> Target //source/exe:envoy-static failed to build
envoy> INFO: Elapsed time: 61.783s, Critical Path: 47.72s
envoy> INFO: 2481 processes: 1040 internal, 1441 local.
envoy> FAILED: Build did NOT complete successfully
```

Looks like `-Wno-error` from `bazelBuildFlags` isn't propagated to that build step:

```
/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/gcc
-U_FORTIFY_SOURCE
-fstack-protector
-Wall
-Wunused-but-set-parameter
-Wno-free-nonheap-object
-fno-omit-frame-pointer
-g0
-O2 '-D_FORTIFY_SOURCE=1'
-DNDEBUG
-ffunction-sections
-fdata-sections '-std=c++14'
-MD
-MF bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_objs/protobuf/map_field.d '-frandom-seed=bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_objs/protobuf/map_field.o' '-DBAZEL_CURRENT_REPOSITORY="com_google_protobuf"'
-iquote external/com_google_protobuf
-iquote bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf
-iquote external/com_google_absl
-iquote bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_absl
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/internal_visibility
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/port
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena_align
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/stubs/_virtual_includes/lite
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena_allocation_policy
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/arena_cleanup
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/string_block
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/varint_shuffle
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/io
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/io_win32
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/third_party/utf8_range/_virtual_includes/utf8_validity
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/third_party/utf8_range/_virtual_includes/utf8_range
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/gzip_stream
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/stubs/_virtual_includes/stubs
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/printer
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/zero_copy_sink
-Ibazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/io/_virtual_includes/tokenizer
-isystem bazel-out/k8-opt-exec-2B5CBBC6/bin/bazel/foreign_cc/zlib/include
-g0 '-frandom-seed=5aycnjj3bc'
-isystem /nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/include
-isystem /nix/store/f1sx0rkgn7zs5h6rvhavwpbw5896cn9y-openjdk-headless-11.0.25+9/include
-isystem /nix/store/pr0rrwgi9fxxkk31628bw8wqfwj66fj2-linux-headers-6.12/include
-isystem /nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/include
-isystem /nix/store/f1sx0rkgn7zs5h6rvhavwpbw5896cn9y-openjdk-headless-11.0.25+9/include
-isystem /nix/store/pr0rrwgi9fxxkk31628bw8wqfwj66fj2-linux-headers-6.12/include
-g0 '-std=c++20'
-fsized-deallocation
-DHAVE_ZLIB
-Woverloaded-virtual
-Wno-sign-compare
-Wno-nonnull
-Werror
-fno-canonical-system-headers
-Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"'
-c external/com_google_protobuf/src/google/protobuf/map_field.cc
-o bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_objs/protobuf/map_field.o)
```

--- a/pkgs/by-name/en/envoy/0006-gcc-warnings.patch
+++ b/pkgs/by-name/en/envoy/0006-gcc-warnings.patch
@@ -1,0 +1,127 @@
+From 448e4e14f4f188687580362a861ae4a0dbb5b1fb Mon Sep 17 00:00:00 2001
+From: "Krinkin, Mike" <krinkin.m.u@gmail.com>
+Date: Sat, 16 Nov 2024 00:40:40 +0000
+Subject: [PATCH] [contrib] Disable GCC warnings and broken features (#37131)
+
+Currently contrib does not build with GCC because of various false
+positive compiler warnings turned to errors and a GCC compiler bug.
+
+Let's first start with the bug, in GCC apparently
+using -gsplit-dwarf (debug fission) and -fdebug-types-section (used to
+optimize the size of debug inforamtion), when used together, can result
+in a linker failure.
+
+Refer to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=110885 for the GCC
+bug report of this issue. When it comes to Envoy, optimized builds with
+GCC are affected on at least GCC 11 (used by --config=docker-gcc) and
+GCC 12 (and I'm pretty sure the bug isn't fixed in any newer versions
+either, though I didn't check each version).
+
+Given that we cannot have both debug fission and a debug types section,
+we decided to abandon the debug types sections and keep the fission.
+
+That being said, apparently both of those options are unmaintained in
+GCC which poses a question of long term viability of using those or GCC.
+
+Other changes in this commit disable GCC compiler errors for various
+warnings that happen when building contrib. I checked those warnings and
+didn't find any true
+positive.
+
+And additionally, for warnings that exists in both Clang and GCC, Clang
+warnings don't trigger, so Clang also disagrees with GCC here.
+
+Additionally missing-requires warning is new and does not exist in GCC
+11, but exists in later versions of GCC, so to avoid breaking on this
+warning for future versions of GCC I disabled it, but also tell GCC to
+not complain if it sees a flag related to an unknwon diagnostic.
+
+This is the last change required to make GCC contrib builds work (you
+can find more context and discussions in
+https://github.com/envoyproxy/envoy/issues/31807)
+
+Risk Level: Low
+Testing: building with --config=gcc and --config=docker-gcc
+Docs Changes: N/A
+Release Notes: N/A
+Platform Specific Features: N/A
+Fixes #31807
+
+Signed-off-by: Mikhail Krinkin <krinkin.m.u@gmail.com>
+---
+ .bazelrc                 | 18 +++++++++++++++++-
+ bazel/envoy_internal.bzl | 16 +++++++++++++++-
+ 2 files changed, 32 insertions(+), 2 deletions(-)
+
+diff --git a/.bazelrc b/.bazelrc
+index e0e4899cecf1..7df94c77944c 100644
+--- a/.bazelrc
++++ b/.bazelrc
+@@ -57,9 +57,9 @@ test --experimental_ui_max_stdouterr_bytes=11712829 #default 1048576
+ # Allow tags to influence execution requirements
+ common --experimental_allow_tags_propagation
+ 
++build:linux --copt=-fdebug-types-section
+ # Enable position independent code (this is the default on macOS and Windows)
+ # (Workaround for https://github.com/bazelbuild/rules_foreign_cc/issues/421)
+-build:linux --copt=-fdebug-types-section
+ build:linux --copt=-fPIC
+ build:linux --copt=-Wno-deprecated-declarations
+ build:linux --cxxopt=-std=c++20 --host_cxxopt=-std=c++20
+@@ -95,6 +95,21 @@ build:gcc --linkopt=-fuse-ld=gold --host_linkopt=-fuse-ld=gold
+ build:gcc --test_env=HEAPCHECK=
+ build:gcc --action_env=BAZEL_COMPILER=gcc
+ build:gcc --action_env=CC=gcc --action_env=CXX=g++
++# This is to work around a bug in GCC that makes debug-types-section
++# option not play well with fission:
++# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=110885
++build:gcc --copt=-fno-debug-types-section
++# These trigger errors in multiple places both in Envoy dependecies
++# and in Envoy code itself when using GCC.
++# And in all cases the reports appear to be clear false positives.
++build:gcc --copt=-Wno-error=restrict
++build:gcc --copt=-Wno-error=uninitialized
++build:gcc --cxxopt=-Wno-missing-requires
++# We need this because -Wno-missing-requires options is rather new
++# in GCC, so flags -Wno-missing-requires exists in GCC 12, but does
++# not in GCC 11 and GCC 11 is what is used in docker-gcc
++# configuration currently
++build:gcc --cxxopt=-Wno-unknown-warning
+ 
+ # Clang-tidy
+ # TODO(phlax): enable this, its throwing some errors as well as finding more issues
+@@ -375,6 +390,7 @@ build:docker-clang-libc++ --config=docker-sandbox
+ build:docker-clang-libc++ --config=rbe-toolchain-clang-libc++
+ 
+ build:docker-gcc --config=docker-sandbox
++build:docker-gcc --config=gcc
+ build:docker-gcc --config=rbe-toolchain-gcc
+ 
+ build:docker-asan --config=docker-sandbox
+diff --git a/bazel/envoy_internal.bzl b/bazel/envoy_internal.bzl
+index 015659851c1b..27ecaa0bbf47 100644
+--- a/bazel/envoy_internal.bzl
++++ b/bazel/envoy_internal.bzl
+@@ -68,7 +68,21 @@ def envoy_copts(repository, test = False):
+                    "-Wc++2a-extensions",
+                    "-Wrange-loop-analysis",
+                ],
+-               repository + "//bazel:gcc_build": ["-Wno-maybe-uninitialized"],
++               repository + "//bazel:gcc_build": [
++                   "-Wno-maybe-uninitialized",
++                   # GCC implementation of this warning is too noisy.
++                   #
++                   # It generates warnings even in cases where there is no ambiguity
++                   # between the overloaded version of a method and the hidden version
++                   # from the base class. E.g., when the two have different number of
++                   # arguments or incompatible types and therefore a wrong function
++                   # cannot be called by mistake without triggering a compiler error.
++                   #
++                   # As a safeguard, this warning is only disabled for GCC builds, so
++                   # if Clang catches a problem in the code we would get a warning
++                   # anyways.
++                   "-Wno-error=overloaded-virtual",
++               ],
+                # Allow 'nodiscard' function results values to be discarded for test code only
+                # TODO(envoyproxy/windows-dev): Replace /Zc:preprocessor with /experimental:preprocessor
+                # for msvc versions between 15.8 through 16.4.x. see

--- a/pkgs/by-name/en/envoy/0007-protobuf-remove-Werror.patch
+++ b/pkgs/by-name/en/envoy/0007-protobuf-remove-Werror.patch
@@ -1,0 +1,19 @@
+diff -Naur a/bazel/protobuf.patch b/bazel/protobuf.patch
+--- a/bazel/protobuf.patch	2025-01-06 23:00:26.683972526 +0100
++++ b/bazel/protobuf.patch	2025-01-07 00:53:33.997482569 +0100
+@@ -149,3 +149,15 @@
+  #if PROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII
+  #define PROTOBUF_DEBUG true
+  #else
++diff -Naur a/build_defs/cpp_opts.bzl b/build_defs/cpp_opts.bzl
++--- a/build_defs/cpp_opts.bzl	2025-01-06 23:02:56.356552216 +0100
+++++ b/build_defs/cpp_opts.bzl	2025-01-07 00:23:30.534047300 +0100
++@@ -22,7 +22,7 @@
++         "-Woverloaded-virtual",
++         "-Wno-sign-compare",
++         "-Wno-nonnull",
++-        "-Werror",
+++        "-Wno-maybe-uninitialized",
++     ],
++ })
++ 

--- a/pkgs/by-name/en/envoy/package.nix
+++ b/pkgs/by-name/en/envoy/package.nix
@@ -30,16 +30,16 @@ let
     # However, the version string is more useful for end-users.
     # These are contained in a attrset of their own to make it obvious that
     # people should update both.
-    version = "1.32.0";
-    rev = "86dc7ef91ca15fb4957a74bd599397413fc26a24";
-    hash = "sha256-Sb87yQXtNbKrqPVujPbfOE+Y8ARmW+HyqMmLLa5tPmA=";
+    version = "1.32.3";
+    rev = "58bd599ebd5918d4d005de60954fcd2cb00abd95";
+    hash = "sha256-5HpxcsAPoyVOJ3Aem+ZjSLa8Zu6s76iCMiWJbp8RjHc=";
   };
 
   # these need to be updated for any changes to fetchAttrs
   depsHash =
     {
-      x86_64-linux = "sha256-qny2l+gIoWCVRZIodd/Tzuj88f/+ajNXeZo/b9XEfVE=";
-      aarch64-linux = "sha256-DkibjmY1YND9Q2aQ41bhNdch0SKM5ghY2mjYSQfV31N=";
+      x86_64-linux = "sha256-YFXNatolLM9DdwkMnc9SWsa6Z6/aGzqLmo/zKE7OFy0=";
+      aarch64-linux = "sha256-AjG1OBjPjiSwWCmIJgHevSQHx8+rzRgmLsw3JwwD0hk=";
     }
     .${stdenv.system} or (throw "unsupported system ${stdenv.system}");
 

--- a/pkgs/by-name/en/envoy/package.nix
+++ b/pkgs/by-name/en/envoy/package.nix
@@ -38,8 +38,8 @@ let
   # these need to be updated for any changes to fetchAttrs
   depsHash =
     {
-      x86_64-linux = "sha256-KQ0ZxLC/ZLLcypmb1UlVXvLWErLmxuednjKRFaBgKuQ=";
-      aarch64-linux = "sha256-DkibjmY1YND9Q2aQ41bhNdch0SKM5ghY2mjYSQfV30M=";
+      x86_64-linux = "sha256-qny2l+gIoWCVRZIodd/Tzuj88f/+ajNXeZo/b9XEfVE=";
+      aarch64-linux = "sha256-DkibjmY1YND9Q2aQ41bhNdch0SKM5ghY2mjYSQfV31N=";
     }
     .${stdenv.system} or (throw "unsupported system ${stdenv.system}");
 
@@ -73,6 +73,18 @@ buildBazelPackage rec {
       # cherry-pick of https://github.com/envoyproxy/envoy/commit/019f589da2cc8da7673edd077478a100b4d99436
       # drop with v1.33.x
       ./0005-deps-Bump-rules_rust-0.54.1-37056.patch
+
+      # patch gcc flags to work with GCC 14
+      # (silences erroneus -Werror=maybe-uninitialized and others)
+      # cherry-pick of https://github.com/envoyproxy/envoy/commit/448e4e14f4f188687580362a861ae4a0dbb5b1fb
+      # drop with v1.33.x
+      ./0006-gcc-warnings.patch
+
+      # Remove "-Werror" from protobuf build
+      # This is fixed in protobuf v28 and later:
+      # https://github.com/protocolbuffers/protobuf/commit/f5a1b178ad52c3e64da40caceaa4ca9e51045cb4
+      # drop with v1.33.x
+      ./0007-protobuf-remove-Werror.patch
     ];
     postPatch = ''
       chmod -R +w .
@@ -206,6 +218,8 @@ buildBazelPackage rec {
       "--noexperimental_strict_action_env"
       "--cxxopt=-Wno-error"
       "--linkopt=-Wl,-z,noexecstack"
+      "--config=gcc"
+      "--verbose_failures"
 
       # Force use of system Java.
       "--extra_toolchains=@local_jdk//:all"

